File: src/cli/colors.rs
// src/cli/colors.rs
use std::env;

/// 颜色支持检测和管理
#[derive(Debug, Clone, Copy, PartialEq)]
pub enum ColorSupport {
    /// 支持彩色输出
    Enabled,
    /// 不支持彩色输出
    Disabled,
}

impl ColorSupport {
    /// 检测当前终端是否支持颜色
    pub fn detect() -> Self {
        // 检查 NO_COLOR 环境变量
        if env::var("NO_COLOR").is_ok() {
            return Self::Disabled;
        }
        
        // 检查 FORCE_COLOR 环境变量
        if let Ok(force) = env::var("FORCE_COLOR") {
            return if force == "0" { Self::Disabled } else { Self::Enabled };
        }
        
        // 检查 TERM 环境变量
        if let Ok(term) = env::var("TERM") {
            if term == "dumb" {
                return Self::Disabled;
            }
        }
        
        // Windows 检查
        #[cfg(windows)]
        {
            if let Ok(ansi) = env::var("ANSICON") {
                if ansi.is_empty() {
                    return Self::Disabled;
                }
            }
        }
        
        // 默认支持颜色（大多数现代终端都支持）
        Self::Enabled
    }
    
    /// 是否支持颜色
    pub fn is_enabled(self) -> bool {
        matches!(self, Self::Enabled)
    }
}

/// ANSI 颜色代码
pub mod ansi {
    pub const RESET: &str = "\x1b[0m";
    pub const BOLD: &str = "\x1b[1m";
    pub const DIM: &str = "\x1b[2m";
    
    // 前景色
    pub const BLACK: &str = "\x1b[30m";
    pub const RED: &str = "\x1b[31m";
    pub const GREEN: &str = "\x1b[32m";
    pub const YELLOW: &str = "\x1b[33m";
    pub const BLUE: &str = "\x1b[34m";
    pub const MAGENTA: &str = "\x1b[35m";
    pub const CYAN: &str = "\x1b[36m";
    pub const WHITE: &str = "\x1b[37m";
    
    // 亮色
    pub const BRIGHT_BLACK: &str = "\x1b[90m";
    pub const BRIGHT_RED: &str = "\x1b[91m";
    pub const BRIGHT_GREEN: &str = "\x1b[92m";
    pub const BRIGHT_YELLOW: &str = "\x1b[93m";
    pub const BRIGHT_BLUE: &str = "\x1b[94m";
    pub const BRIGHT_MAGENTA: &str = "\x1b[95m";
    pub const BRIGHT_CYAN: &str = "\x1b[96m";
    pub const BRIGHT_WHITE: &str = "\x1b[97m";
}

File: src/colors.cpp
#include "pawc/colors.h"

#ifdef _WIN32
#include <io.h>      // for _isatty
#define isatty _isatty
#define fileno _fileno
#else
#include <unistd.h>  // for isatty
#endif

namespace pawc {

// ANSI颜色代码定义
const std::string Colors::RESET = "\033[0m";
const std::string Colors::BOLD = "\033[1m";
const std::string DIM = "\033[2m";

const std::string Colors::RED = "\033[31m";
const std::string Colors::GREEN = "\033[32m";
const std::string Colors::YELLOW = "\033[33m";
const std::string Colors::BLUE = "\033[34m";
const std::string Colors::MAGENTA = "\033[35m";
const std::string Colors::CYAN = "\033[36m";
const std::string Colors::WHITE = "\033[37m";

const std::string Colors::BRIGHT_RED = "\033[91m";
const std::string Colors::BRIGHT_GREEN = "\033[92m";
const std::string Colors::BRIGHT_YELLOW = "\033[93m";
const std::string Colors::BRIGHT_BLUE = "\033[94m";
const std::string Colors::BRIGHT_MAGENTA = "\033[95m";
const std::string Colors::BRIGHT_CYAN = "\033[96m";

// 橙色 (使用 256 色模式)
const std::string Colors::ORANGE = "\033[38;5;208m";

// 检测是否是终端（支持彩色输出）
bool Colors::isTerminal() {
    return isatty(fileno(stderr)) != 0;
}

// 格式化辅助函数
std::string Colors::error(const std::string& msg) {
    if (!isTerminal()) return msg;
    return BOLD + BRIGHT_RED + msg + RESET;
}

std::string Colors::warning(const std::string& msg) {
    if (!isTerminal()) return msg;
    return BOLD + BRIGHT_YELLOW + msg + RESET;
}

std::string Colors::success(const std::string& msg) {
    if (!isTerminal()) return msg;
    return BOLD + GREEN + msg + RESET;  // 使用普通绿色，更柔和
}

std::string Colors::info(const std::string& msg) {
    if (!isTerminal()) return msg;
    return BOLD + CYAN + msg + RESET;  // 使用青色，比蓝色更温和
}

std::string Colors::highlight(const std::string& msg) {
    if (!isTerminal()) return msg;
    return BOLD + YELLOW + msg + RESET;  // 使用黄色，比洋红色柔和
}

std::string Colors::dimmed(const std::string& msg) {
    if (!isTerminal()) return msg;
    return DIM + msg + RESET;
}

std::string Colors::orange(const std::string& msg) {
    if (!isTerminal()) return msg;
    return BOLD + ORANGE + msg + RESET;
}

} // namespace pawc

